src/components/routes/ViewConfig.tsx

```diff
@@ -37,15 +37,15 @@ const Section = styled.div`
 const ViewConfig: SFC<GraphProps> = props => {
     const onSubmit = () => {
         const { dataSetId, collection } = props.metadata.dataSetMetadata!;
-        const viewConfig = props.denormalizeTree();
 
-        if (typeof viewConfig === 'string') {
-            return alert(viewConfig); // TODO: Make this fancy, I'd suggest to maybe add an optional error to NormalizedComponentConfig, add a scrollTo and style the selectBox accordingly
+        try {
```

Comment:
Mutate errors don't get handled. (no `.catch`)
I'd just add `await` before the props.mutate call, that way you can catch both sync & async errors in the same catch block.